test/e2e/e2e_suite_test.go

```diff
@@ -52,6 +52,8 @@ var (
 
 // Test suite global vars
 var (
+	ctx = context.Background()
```

Comment:
should this somehow be using `withCancel` that cancelI() on syscall.SIGINT, syscall.SIGTERM?

Reply:
This PR was mostly to avoid spreading of new context in our suite (see e.g https://github.com/kubernetes-sigs/cluster-api/pull/4038#discussion_r642358800), so it will be simpler to implement something like you are suggesting; we discussed also to have an overall test timeout, but this is a little bit trickier given that we are running different combination of tests.

Reply:
Can we use https://github.com/kubernetes-sigs/cluster-api/blob/2e4deb4e6c202ce4c9ca85df6996b0545f4fccce/util/suite_test.go#L27 ?

Reply:
Done!